content/documentation/tutorials/examples/async-sub.go

```diff
@@ -2,29 +2,36 @@
 
 package main
 
-// Import packages
+// Import Go and NATS packages
 import (
 	"log"
+	"os"
 	"runtime"
 
-	"github.com/nats-io/go-nats"
-)
+	"github.com/nats-io/nats"
+) 
 
 func main() {
 
-	// Create server connection: auth and no auth
-	// natsConnection, _ := nats.Connect("nats://foo:bar@localhost:4222")
-	natsConnection, _ := nats.Connect(nats.DefaultURL)
-	log.Println("Connected to " + nats.DefaultURL)
-
-	// Subscribe to subject
-	log.Printf("Subscribing to subject 'foo'\n")
-	natsConnection.Subscribe("foo", func(msg *nats.Msg) {
-
-		// Handle the message
-		log.Printf("Received message '%s\n", string(msg.Data)+"'")
-	})
-
-	// Keep the connection alive
-	runtime.Goexit()
-}
+    // Create server connection: auth and no auth
+    // natsConnection, _ := nats.Connect("nats://foo:bar@localhost:4222")
+    natsConnection, _ := nats.Connect(nats.DefaultURL)
```

Comment:
Not to your making, but not checking for error here is a bit wrong because if user forgets to run the NATS Server, this program will not complain but will not receive any message. Hopefully, the sender program will complain about not  being able to connect ;-)